golos.ctrl/src/golos.ctrl.cpp

```diff
@@ -55,6 +59,54 @@ void control::updateprops(properties new_props) {
     });
 }
 
+void control::transfer(account_name from, account_name to, asset quantity, string memo) {
+    if (to == _self && quantity.amount > 0) {
+        // Don't check `from` for now, just distribute to top witnesses
+
+        //Reinitialize props because called without symbol
+        _token = quantity.symbol;
+        props_tbl tbl{_self, _token};
+        auto i = tbl.begin();
+        bool exists = i != tbl.end();
+        if (!exists) {
+            print("symbol not found");
+            return;     // it's notification, so fail silently
+        }
+        _props = i->props;
+        _owner = i->owner;
+        _has_props = exists;
+
+        // Distribute between top witnesses
+        auto total = quantity.amount;
+        auto top = get_top_witnesses();
+        auto n = top.size();
+        if (n == 0) {
+            print("nobody in top");
+            return;
```

Comment:
In this case, the contract will accumulate unspent funds. Should we send them out later?

Reply:
We have no ready solution on this, propose implement it in separate issue

Reply:
While the community is not running, it does not distribute funds among witnesses. I think this is not a problem. In any case, delegates can distribute funds from this account by forming the appropriate transaction.